### PR TITLE
dotCMS/core#14183 Make #dotedit works in new edit mode

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 1]
```diff
@@ -187,6 +199,22 @@ export class DotEditContentHtmlService {
         return this.getEditPageIframe().contentWindow.getDotNgModel();
     }
 
+    private handleIframeClicks(target: HTMLElement) {
+        if (target.dataset.dotObject === 'edit-content') {
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: this 'edit-content' shouls be a const somewhere

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 2]
```diff
@@ -373,8 +411,8 @@ export class DotEditContentHtmlService {
             save: (contentletEvent: any) => {
                 if (this.currentAction === DotContentletAction.ADD) {
                     this.renderAddedContentlet(contentletEvent.data);
-                } else if (this.currentAction === DotContentletAction.EDIT) {
-                    this.renderEditedContentlet(contentletEvent.data);
+                } else {
+                    this.renderEditedContentlet(this.currentContentlet || contentletEvent.data);
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: this should be something like:

                if (this.currentAction === DotContentletAction.ADD) {
                    data = contentletEvent.data;
                } else  {
                    data = this.currentContentlet || contentletEvent.data;
                } 
                this.renderEditedContentlet(this.currentContentlet || contentletEvent.data);

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.spec.ts [hunk 3]
```diff
@@ -147,7 +147,28 @@ describe('DotEditContentHtmlService', () => {
...
+            }
+        });
+
+        expect(this.dotEditContentHtmlService.renderEditedContentlet).toHaveBeenCalledWith({
+            identifier: '444',
+            inode: '555'
+        });
+    });
+
+    it('should render relocated contentlet', () => {
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.spec.ts]: don't we need a test to check this?

https://github.com/dotCMS/core-web/pull/667/files#diff-e2760b805d923abc4137bf2e3ddf0ee9R66

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts (hunk 1 above)
Comment (fmontes) [reply src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: When we do the refactor we talked about it on how to handle all the clicks in the iframe.window we do it, right now is just one action that is being handle it like this.

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts (hunk 2 above)
Comment (fmontes) [reply src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: renderAddedContentlet and renderEditedContentlet are different methods.

File: src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts [hunk 4]
```diff
@@ -187,6 +199,23 @@ export class DotEditContentHtmlService {
         return this.getEditPageIframe().contentWindow.getDotNgModel();
     }
 
+    private handleIframeClicks(target: HTMLElement) {
+        console.log('click iframe', target);
```
Comment (hmoreras) [inline src/app/portlets/dot-edit-page/content/services/dot-edit-content-html/dot-edit-content-html.service.ts]: remove console.log.

**Pull request merged**